backends/dpdk/dpdkMetadata.cpp

```diff
@@ -42,23 +42,25 @@ void DirectionToRegRead::uniqueNames(IR::DpdkAsmProgram *p) {
     }
     reg_read_tmp = mng.newName("reg_read_tmp");
     left_shift_tmp = mng.newName("left_shift_tmp");
-    registerInstanceName = "network_port_mask";
 
-    P4::MinimalNameGenerator mng1;
-    for (auto decl : p->externDeclarations) {
-        mng1.usedName(decl->name);
-    }
     // "network_port_mask" name is used in dpdk for initialzing direction port mask
     // make sure no such decls exist with that name
-    if (mng1.newName(registerInstanceName) != registerInstanceName)
+    registerInstanceName = "network_port_mask";
+    for (auto decl : p->externDeclarations) {
+        usedNames.insert(decl->name);
+    }
+
+    if (usedNames.count(registerInstanceName))
```

Comment:
This works if usedNames contains everything you need to check for. I was hoping that you'd add a new method `MinimalNameGenerator::nameIsUsed(cstring name)`